app/Console/Commands/KeyGenerateCommand.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Console\Commands;
+
+use Illuminate\Encryption\Encrypter;
+use Safe\Exceptions\UrlException;
+
+class KeyGenerateCommand extends \Illuminate\Foundation\Console\KeyGenerateCommand
+{
+	/**
+	 * Set the application key in the environment file.
+	 *
+	 * @param string $key
+	 *
+	 * @return bool
+	 *
+	 * @throws UrlException
+	 */
+	protected function setKeyInEnvironmentFile($key): bool
+	{
+		$currentKey = $this->laravel['config']['app.key'];
+		if (str_starts_with($currentKey, 'base64:')) {
+			$currentKey = substr($currentKey, 7);
+		}
+		$supported = Encrypter::supported(\Safe\base64_decode($currentKey), $this->laravel['config']['app.cipher']);
+
+		if (strlen($currentKey) !== 0 && ($supported || ($this->getDefaultConfirmCallback()() && !$this->confirmToProceed()))) {
```

Comment:
Why do you call `getDefaultConfirmCallback` explicitly`? It is called inside `confirmToProceed` anyway. The original condition from the parent class does not do it either. There it is only:

```php
if (strlen($currentKey) !== 0 && (! $this->confirmToProceed())) {
```

What do you want to achieve with that? If there is a non-obvious reason you should add the explanation to the PhpDoc comment above the method and explain what this method does differently than the original and why. Other developers may wonder the same in the future.